compiler-rt/lib/tsan/rtl/tsan_platform_linux.cpp

```diff
@@ -484,6 +485,7 @@ int ExtractRecvmsgFDs(void *msgp, int *fds, int nfd) {
   return res;
 }
 
+#    if !SANITIZER_ANDROID
```

Comment:
it seems like you should probably add the android case _inside_ this function, rather than adding a special case just for Android on line 573?